lib/urlwatch/worker.py

```diff
@@ -41,64 +41,75 @@
 MAX_WORKERS = 10
 
 
-def run_parallel(func, items):
+def run_jobs(urlwatcher):
+    cache_storage = urlwatcher.cache_storage
+    jobs = urlwatcher.jobs
+    report = urlwatcher.report
+
+    logger.debug('Processing %d jobs', len(jobs))
+    main_thread_jobs = [job for job in jobs if job.MAIN_THREAD]
```

Comment:
These two lines might be pre-destined for https://stackoverflow.com/q/12720151/1047040 (any of the solutions, not necessarily mine ;), but maybe it's okay as-is, your choice :)